go/vt/mysqlctl/builtinbackupengine.go

```diff
@@ -182,7 +187,9 @@ func (be *BuiltinBackupEngine) ExecuteBackup(ctx context.Context, params BackupP
 	params.Logger.Infof("using replication position: %v", replicationPosition)
 
 	// shutdown mysqld
-	err = params.Mysqld.Shutdown(ctx, params.Cnf, true)
+	shutdownCtx, cancel := context.WithTimeout(ctx, *builtinBackupMysqldDeadline)
+	err = params.Mysqld.Shutdown(shutdownCtx, params.Cnf, true)
```

Comment:
I don't see an easy way to test this. I can't use `fakemysqldaemon`, because that package imports this package, which causes circular import errors. 🤔

Reply:
Solved this with blackbox testing ✅